authentication/auth_method_azure_cli_token.go

```diff
@@ -182,21 +196,53 @@ func obtainAuthorizationToken(endpoint string, subscriptionId string) (*cli.Toke
 	return &token, nil
 }
 
-// obtainSubscription return a subscription object of the specified subscriptionId.
-// If the subscriptionId is empty, it returns the default subscription.
+// obtainSubscription returns a Subscription object of the specified subscriptionId.
+// If the subscriptionId is empty, it selects the default subscription.
 func obtainSubscription(subscriptionId string) (*cli.Subscription, error) {
-	var sub cli.Subscription
+	var acc cli.Subscription
 	cmd := make([]string, 0)
 	cmd = []string{"account", "show", "-o=json"}
 	if subscriptionId != "" {
 		cmd = append(cmd, "-s", subscriptionId)
 	}
-	err := jsonUnmarshalAzCmd(&sub, cmd...)
+	err := jsonUnmarshalAzCmd(&acc, cmd...)
 	if err != nil {
 		return nil, fmt.Errorf("Error parsing json result from the Azure CLI: %v", err)
 	}
 
-	return &sub, nil
+	return &acc, nil
+}
+
+// obtainTenant returns a Subscription object having the specified tenantId.
+// If the tenantId is empty, it selects the default subscription.
+// This works with `az login --allow-no-subscriptions`
+func obtainTenant(tenantId string) (*cli.Subscription, error) {
+	var acc cli.Subscription
+	if tenantId == "" {
+		cmd := make([]string, 0)
+		cmd = []string{"account", "show", "-o=json"}
+		err := jsonUnmarshalAzCmd(&acc, cmd...)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing json result from the Azure CLI: %v", err)
+		}
+	} else {
+		var accs []cli.Subscription
+		cmd := make([]string, 0)
+		cmd = []string{"account", "list", "-o=json"}
+		err := jsonUnmarshalAzCmd(&accs, cmd...)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing json result from the Azure CLI: %v", err)
+		}
+
+		for _, a := range accs {
+			if a.TenantID == tenantId {
+				acc = a
+				break
+			}
+		}
+	}
```

Comment:
edge-case but what if this isn't found - should we raise an error here?

Reply:
Good call, not necessarily an edge case

Reply:
An error should now be raised when the specified tenantId is not found in az accounts